### PR TITLE
PeptideIndexer error fixed. 

### PR DESCRIPTION
**Pull request by ypriverol**
## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated
- [ ] If necessary, also make a PR on the [nf-core/proteomicslfq branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/proteomicslfq)

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -709,6 +712,7 @@ process index_peptides {
                     -enzyme:specificity ${pepidx_num_enzyme_termini} \\
                     ${il} \\
                     ${allow_um} \\
+                    ${unmatch_action} \\
```
Comment (jpfeuffer) [inline main.nf:715]:
```suggestion
                    -unmatched_action ${unmatch_action} \\
```

File: nextflow_schema.json [hunk 2]
```diff
@@ -510,6 +510,17 @@
...
                 },
+                "unmatched_action": {
+                    "type": "string",
+                    "description": "Action to be taken if peptide sequences cannot be matched to any protein: 1) raise an error; 2) warn (unmatched PepHits will miss target/decoy annotation with downstream problems); 3) remove the hit. (default: 'error' valid: 'error', 'warn', 'remove')",
+                    "default": "error",
+                    "fa_icon": "far fa-check-square",
+                    "enum": [
+                        "error",
+                        "warn",
+                        "remove"
```
Comment (jpfeuffer) [inline nextflow_schema.json:521]: You do not support "remove" in your ifcase in the main.nf
Please add it. You do not need the "def if case" actually.

File: nextflow_schema.json [hunk 3]
```diff
@@ -510,6 +510,17 @@
                         "true"
                     ]
                 },
+                "unmatched_action": {
```
Comment (jpfeuffer) [inline nextflow_schema.json:513]: Can you make it hidden = true? I.e. an advanced parameter?

**Review by jpfeuffer — CHANGES_REQUESTED** [review]
While this is a nice addition, I think in such cases the reason for unmatched peptides should be investigated first.
In proteogenomics (e.g. point mutations) for example the mismatches_max or max_aaa parameters should be tweaked first. Maybe you can add them, too as advanced params.

File: nextflow_schema.json (hunk 2 above)
Comment (ypriverol) [reply nextflow_schema.json:521]: Should we support remove?

Comment (ypriverol) [reply nextflow_schema.json:521]: Supported !!!!

**Comment by ypriverol** [comment]
> While this is a nice addition, I think in such cases the reason for unmatched peptides should be investigated first.
> In proteogenomics (e.g. point mutations) for example the mismatches_max or max_aaa parameters should be tweaked first. Maybe you can add them, too as advanced params.

Following this topic, the Proteogenomics database contains variants which one/two aminoacids difference but we also added the Protein accession of the `Variant protein` meaning: 

```
> Reference Protein
AAAAAAAAAAK
> Variant Protein
AAAACAAAAAK
```

In the example, both proteins are in the database file. I suspect if both peptides are identified, the PeptideIndexer should be able to match Peptide 1 `AAAAAAAAAAK` -> `Reference Protein` and Peptide 2 `AAAACAAAAAK` -> `Variant Protein`. I actually thought that PeptideIndexer unmatched will only happen when a search engine uses a fancy mechanism to detect variants and start reporting it and you can not match any of those variants to the database.

File: environment.yml [hunk 4]
```diff
@@ -10,7 +10,8 @@ dependencies:
   - openms::openms=2.7.0pre
   - openms::openms-thirdparty=2.7.0pre # 08.02.2021
   - bioconda::bioconductor-msstats=3.20.1 # will include R
-  - bioconda::sdrf-pipelines=0.0.13 # for SDRF conversion
+  - bioconda::sdrf-pipelines=0.0.14 # for SDRF conversion
+  - bioconda::thermorawfileparser=1.2.3
```
Comment (jpfeuffer) [inline environment.yml:14]: TRFP: this should be done in the openms-thirdparty conda package. I will make sure to do it at the next update.

File: conf/base.config [hunk 5]
```diff
@@ -37,8 +37,8 @@ process {
     time = { check_max( 6.h * task.attempt, 'time' ) }
   }
   withLabel:process_medium {
-    cpus = { check_max( 8 * task.attempt, 'cpus' ) }
-    memory = { check_max( 32.GB * task.attempt, 'memory' ) }
+    cpus = { check_max( 2 * task.attempt, 'cpus' ) }
```
Comment (jpfeuffer) [inline conf/base.config:40]: This should be either reverted or all the labels adapted (such that low is actually lower than medium). Or add more labels that tweak time only, if that is what you wanted to do.

File: nextflow_schema.json [hunk 6]
```diff
@@ -510,6 +510,18 @@
...
+                    "type": "string",
+                    "description": "Action to be taken if peptide sequences cannot be matched to any protein: 1) raise an error; 2) warn (unmatched PepHits will miss target/decoy annotation with downstream problems); 3) remove the hit. (default: 'error' valid: 'error', 'warn', 'remove')",
+                    "default": "error",
+                    "fa_icon": "far fa-check-square",
```
Comment (jpfeuffer) [inline nextflow_schema.json:517]: I usually use check-square for bool options only. use a list icon or so.

**Review by jpfeuffer — CHANGES_REQUESTED** [review]

File: main.nf [hunk 7]
```diff
@@ -568,6 +569,7 @@ process index_peptides {
                     -enzyme:specificity ${pepidx_num_enzyme_termini} \\
                     ${il} \\
                     ${allow_um} \\
+                    -unmatched_action ${unmatch_action} \\
```
Comment (jpfeuffer) [inline main.nf:572]:
```suggestion
                    -unmatched_action ${params.unmatched_action} \\
```

**Review by jpfeuffer — CHANGES_REQUESTED** [review]

File: environment.yml [hunk 8]
```diff
@@ -10,12 +10,13 @@ dependencies:
...
-  - bioconda::sdrf-pipelines=0.0.13 # for SDRF conversion
+  - bioconda::sdrf-pipelines=0.0.14 # for SDRF conversion
   - conda-forge::r-ptxqc=1.0.5 # for QC reports
   - conda-forge::xorg-libxt=1.2.0 # until this R fix is merged: https://github.com/conda-forge/r-base-feedstock/pull/128
   - conda-forge::fonts-conda-ecosystem=1 # for the fonts in QC reports
   - conda-forge::python=3.8.5
   - conda-forge::markdown=3.2.2
   - conda-forge::pymdown-extensions=8.0.1
   - conda-forge::pygments=2.7.1
+  - conda-forge::mono=5.14.0.177
```
Comment (jpfeuffer) [inline environment.yml:21]:
```suggestion
  - conda-forge::mono=5.14.0.177
  - conda-forge::pandoc=2.11.4
```

Thanks conda for another incompatibility error. Wherever this comes from?!

Comment (jpfeuffer) [reply environment.yml:21]: Just released yesterday, already breaks things.

**Review by jpfeuffer — APPROVED** [review]

**Pull request merged**